tracer/src/Datadog.Trace/Logging/LogsInjection/LibLogScopeEventSubscriber.cs

```diff
@@ -37,17 +34,7 @@ internal class LibLogScopeEventSubscriber : IDisposable
         private readonly ILogEnricher _logEnricher;
 
         private readonly AsyncLocal<IDisposable> _currentEnricher = new();
-
-        // Each mapped context sets a key-value pair into the logging context
-        // Disposing the returned context unsets the key-value pair
-        // Keep a stack to retain the history of our correlation identifier properties
-        // (the stack is particularly important for Serilog, see below).
-        //
-        // IMPORTANT: Serilog -- The logging contexts (throughout the entire application)
-        //            are maintained in a stack, as opposed to a map, and must be closed
-        //            in reverse-order of opening. When operating on the stack-based model,
-        //            it is only valid to add the properties once and unset them once.
-        private readonly ConcurrentStack<IDisposable> _contextDisposalStack = new ConcurrentStack<IDisposable>();
+        private readonly AsyncLocal<Context> _currentContext = new();
```

Comment:
Since the only log provider that will access `_currentContext` is the `FallbackNLogLogProvider` for NLog versions less than 4.0.1 and the library stores its context in thread locals, would it be safe for us to do the same? Might that help with performance for a little bit? I'm also willing to take a performance degradation here for a solution that finally works, not to mention we have plans to improve this with NLog automatic instrumentation

ThreadLocal usage: https://github.com/NLog/NLog/blob/4.0.1/src/NLog/MappedDiagnosticsContext.cs#L50